src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+            error code
+
+    @note If the account includes a regular key or master key, it is included
+   with the signer's list with a maximum weight. If the account does not include
+   a signer's list, the threshold is set to 1 (i.e. either the master key or
+   regular key can sign)
+*/
+std::tuple<std::unordered_map<AccountID, std::uint32_t>, std::uint32_t, TER>
+getSignersListAndQuorum(ApplyView& view, SLE const& sleB, beast::Journal j)
+{
+    std::unordered_map<AccountID, std::uint32_t> r;
+    std::uint32_t q = std::numeric_limits<std::uint32_t>::max();
+
+    AccountID const thisDoor = sleB[sfAccount];
+    auto const sleDoor = [&] { return view.read(keylet::account(thisDoor)); }();
+
+    if (!sleDoor)
+    {
+        return {r, q, tecINTERNAL};  // LCOV_EXCL_LINE
+    }
+
+    auto const masterKey = [&]() -> std::optional<AccountID> {
+        if (sleDoor->isFlag(lsfDisableMaster))
+            return std::nullopt;
+        return thisDoor;
+    }();
+
+    std::optional<AccountID> regularKey = (*sleDoor)[~sfRegularKey];
+
+    auto const sleS = view.read(keylet::signers(sleB[sfAccount]));
+    if (!sleS)
+    {
+        if (masterKey || regularKey)
+        {
+            q = 1;
+            if (masterKey)
+                r[*masterKey] = std::numeric_limits<std::uint16_t>::max();
+            if (regularKey)
+                r[*regularKey] = std::numeric_limits<std::uint16_t>::max();
+
+            return {std::move(r), q, tesSUCCESS};
+        }
+        return {r, q, tecXCHAIN_NO_SIGNERS_LIST};
+    }
+    q = (*sleS)[sfSignerQuorum];
+
+    auto const accountSigners = SignerEntries::deserialize(*sleS, j, "ledger");
+
+    if (!accountSigners)
+    {
+        return {r, q, tecINTERNAL};  // LCOV_EXCL_LINE
+    }
+
+    for (auto const& as : *accountSigners)
+    {
+        r[as.account] = as.weight;
+    }
+
+    // add the master and regular keys. If they are already part of the signer's
+    // list, overwrite their weights.
+    if (masterKey)
+        r[*masterKey] = std::numeric_limits<std::uint16_t>::max();
+    if (regularKey)
+        r[*regularKey] = std::numeric_limits<std::uint16_t>::max();
+
+    return {std::move(r), q, tesSUCCESS};
+};
+}  // namespace
+//------------------------------------------------------------------------------
+
+NotTEC
+BridgeCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;  // LCOV_EXCL_LINE
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const account = ctx.tx[sfAccount];
+    auto const reward = ctx.tx[sfSignatureReward];
+    auto const minAccountCreate = ctx.tx[~sfMinAccountCreateAmount];
+    auto const bridge = ctx.tx[sfXChainBridge];
+    if (bridge.lockingChainDoor() == bridge.issuingChainDoor())
+    {
```

Comment:
Why the two door accounts cannot be the same, for IOUs?

Reply:
Fixed in 91de824b2 [FOLD] Address PR feedback 1: